bcgov/cas-registration
# Task/nx deploy

Addresses bcgov/cas-reporting#165 - [Project link](https://github.com/orgs/bcgov/projects/123/views/1?pane=issue&itemId=54817608)
Adds Nx Container building and automation to add images to artifact registry. Utilizing the make target to deploy to a testing namespace.

## Changes 🚧

- Added a Nx build (`nx build reporting`) target for the reporting frontend.
- Added a Nx target (`nx container reporting`) to containerize the reporting frontend with Docker.
- Added helm templates to handle the new reporting frontend (conditionally).
  - Updated template metadata names to use `_helper.fullname` in all instances. This will stay consistent with `cas-registration` for the core develop branch, but use the `.values.fullnameOverride` in our _giraffe-develop_ branch
- Added github workflow to run artefact generation from Nx for reporting frontend.

## To test 🔬

> You may need to run `yarn` in the `./bciers` directory to install and update packages.
- Navigate to `./bciers` and run (if nx is not installed globally, first `npx `)`nx container reporting --dry-run`. Next.JS should be built and then a container made from that build.
- Run `make install_giraffe --dry-run` against our testing namespace. 
- Check that GitHub actions complete.

## Notes📝 

- Parts of the GitHub Actions CI will need to be re-written once `giraffe-develop` gets merged into the core `develop`. 
- As more of the existing applications get brought into the monorepo tooling, we'll be able to reduce the existing CI builds and handle them with Nx instead. 
  - There's a chance we might have to re-think how we handle tagging, purely due to the fact that Nx can be used to only build what changes. We generally build _everything_ and tag it.
- I've documented some possible optimizations from Nx we could utilize, will share those in https://github.com/orgs/bcgov/projects/123/views/1?filterQuery=-status%3Adone+nx&pane=issue&itemId=55856106 for future investigation
